python/cudf/cudf/dataframe/column.py

```diff
@@ -80,10 +81,10 @@ def _concat(cls, objs, dtype=None):
 
         # Handle categories for categoricals
         if all(isinstance(o, CategoricalColumn) for o in objs):
-            new_cats = tuple(
-                set([val for o in objs for val in o.cat().categories])
-            )
-            objs = [o.cat()._set_categories(new_cats) for o in objs]
+            cats = [o.cat().categories for o in objs]
+            cats = [as_column(cats) for cats in cats]
```

Comment:
This line is super confusing, can you clean it up?

Reply:
How's this look? https://github.com/rapidsai/cudf/blob/1b9350460296abc613aa608dd6ad624a4a7c8cad/python/cudf/cudf/dataframe/column.py#L82-L84

Reply:
Much clearer, resolving.